File: scripts/test-backend.js
const https = require('https');

function get(path) {
  return new Promise((resolve, reject) => {
    const url = 'https://backendmenu-3.onrender.com' + path;
    console.log('\n===== GET ' + path + ' =====');
    https.get(url, (res) => {
      let data = '';
      res.on('data', (chunk) => { data += chunk; });
      res.on('end', () => {
        console.log('Status:', res.statusCode);
        try {
          const json = JSON.parse(data);
          if (Array.isArray(json)) {
            console.log('Array with ' + json.length + ' items');
            if (json.length > 0) {
              console.log('First item keys:', Object.keys(json[0]).join(', '));
              console.log('First item:', JSON.stringify(json[0], null, 2));
            }
            if (json.length > 1) {
              console.log('Second item:', JSON.stringify(json[1], null, 2));
            }
          } else if (json && typeof json === 'object') {
            console.log('Object keys:', Object.keys(json).join(', '));
            if (json.data && Array.isArray(json.data)) {
              console.log('json.data: Array with ' + json.data.length + ' items');
              if (json.data.length > 0) {
                console.log('First data item:', JSON.stringify(json.data[0], null, 2));
              }
            } else {
              console.log('Response:', JSON.stringify(json, null, 2).substring(0, 800));
            }
          }
        } catch (e) {
          console.log('Raw (500 chars):', data.substring(0, 500));
        }
        resolve();
      });
    }).on('error', (err) => {
      console.log('ERROR:', err.message);
      resolve();
    });
  });
}

async function main() {
  console.log('Testing backend: https://backendmenu-3.onrender.com');
  await get('/api/orders');
  await get('/api/menu');
  await get('/api/bills');
  await get('/api/commandes');
}

main();

File: scripts/test-backend.py
import urllib.request
import json

BASE = "https://backendmenu-3.onrender.com"
endpoints = ["/api/orders", "/api/menu", "/api/bills", "/api/commandes"]

for path in endpoints:
    url = BASE + path
    print(f"\n===== GET {path} =====")
    try:
        req = urllib.request.Request(url, headers={"User-Agent": "Mozilla/5.0"})
        with urllib.request.urlopen(req, timeout=30) as resp:
            status = resp.status
            data = resp.read().decode("utf-8")
            print(f"Status: {status}")
            try:
                j = json.loads(data)
                if isinstance(j, list):
                    print(f"Array with {len(j)} items")
                    if len(j) > 0:
                        print(f"First item keys: {list(j[0].keys())}")
                        print(f"First item: {json.dumps(j[0], indent=2, ensure_ascii=False)[:800]}")
                    if len(j) > 1:
                        print(f"Second item: {json.dumps(j[1], indent=2, ensure_ascii=False)[:800]}")
                elif isinstance(j, dict):
                    print(f"Object keys: {list(j.keys())}")
                    if "data" in j and isinstance(j["data"], list):
                        print(f"json.data: Array with {len(j['data'])} items")
                        if len(j["data"]) > 0:
                            print(f"First data item: {json.dumps(j['data'][0], indent=2, ensure_ascii=False)[:800]}")
                    else:
                        print(f"Response: {json.dumps(j, indent=2, ensure_ascii=False)[:800]}")
            except json.JSONDecodeError:
                print(f"Raw text: {data[:500]}")
    except Exception as e:
        print(f"ERROR: {e}")
